packages/rocketchat-ui-sidenav/client/sidebarHeader.js

```diff
@@ -73,6 +73,7 @@ const toolbarButtons = (user) => [{
 {
 	name: t('Directory'),
 	icon: 'globe',
+	condition: () => RocketChat.settings.get('UI_DisplayDirectory'),
 	action: () => {
 		menu.close();
```

Comment:
Consider updating the directory page access to verify this setting to ensure users cannot access it via URL when it’s disabled in the UI.